File: XMLCheck/Handler/CheckRulesXML13/Xml13RuleRunner.cs
using XMLCheck.Models;
using XMLCheck.Services;

namespace XMLCheck.Handler.CheckRulesXML13
{
    public static class Xml13RuleRunner
    {
        private static readonly List<IXml13Rule> _rules = new()
        {

        };
        public static List<ErrorDetails> Run(Xml13Model? model,HoSoContext context)
        {
            var errors = new List<ErrorDetails>();

            foreach (var rule in _rules)
            {
                var err = rule.Check(model,context);
                if (err != null)
                    errors.Add(err);
            }

            return errors;
        }
    }
}

File: XMLValidator_v2/src/parsers/rules/xml13/xml13RuleRunner.ts
import { Xml13Model } from '../../../interfaces/xml-models.interface';
import { HoSoContext } from '../../../services/hoso-context';
import { ErrorDetails } from '../../../dto/xml-check.dto';
import { IXml13Rule } from './xml13RuleBase';

export class Xml13RuleRunner {
  private static readonly rules: IXml13Rule[] = [];

  public static run(model: Xml13Model, context: HoSoContext): ErrorDetails[] {
    const errors: ErrorDetails[] = [];
    if (!model) return errors;

    for (const rule of this.rules) {
      if (!rule.isEnabled) continue;
      const err = rule.check(model, context);
      if (err) {
        errors.push(err);
      }
    }

    return errors;
  }
}
